src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,264 @@ public static class NUnitTestExtractorApp
-                case Level.TestCase:
-                    if (testCase != null)
-                    {
-                        data = FormattedTestCase(methodInfo.DeclaringType.ToString(), methodInfo.Name, testCase.Arguments, data);
-                    }
+            foreach (var item in GetValidTestSuiteTypesFromAssembly(assemblyFullPath))
+            {
+                Level lv = ParseLevel(_options.Level);
 
-                    break;
+                switch (lv)
+                {
+                    case Level.Namespace:
+                        rv.Add(item.Namespace);
+                        break;
+                    case Level.Class:
+                        rv.Add(item.FullName);
+                        break;
+                    case Level.Function:
+                        rv.UnionWith(GetValidTestsFromTestSuite(item));
+                        break;
+                    case Level.TestCase:
+                        break;
+                    default:
+                        throw new ArgumentException(StringUtils.FormatInvariant("Invalid level option: {0}", lv.ToString()));
+                }
             }
 
-            return data;
+            OutputInfo(rv);
         }
 
         /// <summary>
-        /// Add testCase categories to categoryNames 
+        /// Load assembly by passing its path
         /// </summary>
-        /// <param name="categoryNames">list to add to</param>
-        /// <param name="testCase">testCase used to get arguments from</param>
-        private static void AddTestCaseCategoriesToList(List<string> categoryNames, TestCaseAttribute testCase)
+        /// <param name="assemblyFullPath"></param>
+        /// <returns></returns>
+        [SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
+        private static void LoadAssembly(string assemblyFullPath)
         {
-            foreach (string category in testCase.Categories)
+            ThrowIf.StringIsNullOrWhiteSpace(assemblyFullPath, nameof(assemblyFullPath));
+
+            try
             {
-                categoryNames.Add(category);
+                _assembly = Assembly.LoadFrom(assemblyFullPath);
+            }
+            catch (FileNotFoundException)
+            {
+                string errorMessage = StringUtils.FormatInvariant("File was not found: '{0}'", assemblyFullPath);
+
+                throw new FileNotFoundException(errorMessage);
             }
         }
 
         /// <summary>
-        /// Add method categories to list
+        /// Validate whether the type object is a valid test suite
         /// </summary>
-        /// <param name="categoryNames">list to add to</param>
-        /// <param name="categories">categories to add to list</param>
-        private static void AddMethodCategoriesToList(List<string> categoryNames, IEnumerable<Attribute> categories)
+        /// <param name="type"></param>
+        /// <returns></returns>
```

Comment:
[Should] Fill in the param & returns tags.
Same for other functions below that are missing those details.